.github/workflows/test.yml

```diff
@@ -34,3 +36,26 @@ jobs:
       - run: cargo test
       - name: Check semver
         uses: obi1kenobi/cargo-semver-checks-action@v2
+
+  msrv:
+    name: Test MSRV
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: Swatinem/rust-cache@v2
+        if: github.event_name != 'release' && github.event_name != 'workflow_dispatch'
+      - name: Read crate metadata
+        id: metadata
+        run: echo "rust-version=$(sed -ne 's/rust-version *= *\"\(.*\)\"/\1/p' Cargo.toml)" >> $GITHUB_OUTPUT
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: ${{ steps.metadata.outputs.rust-version }}
+          components: clippy,rustfmt
```

Comment:
Sure, that could also work.  The thing with `hack` is that its a fairly complex all-included utility, whereas what we need here is a really dumb way to extract an older version text, switch back to it, and just run regular tests. With hack, we don't know all the things it may do (or forget to do)... Not that important though, so up to you

Reply:
Yeah I don't really care either way; just curious 😂 and eventually this will get built into cargo.

Feel free to merge when the latest CI run succeeds.